src/clegoues/genprog4java/mut/edits/java/JavaEditFactory.java

```diff
@@ -122,53 +109,58 @@ public JavaEditOperation makeEdit(Mutation edit, Location dst, EditHole sources)
 			// different from what is now at that location.
 			// this comes down to our having overloaded statement IDs to mean both location and statement ID
 			// which is a problem I keep meaning to solve.
-			if(mut != Mutation.APPEND && faultAST.equals(fixAST)) {
+			if (mut != Mutation.APPEND && faultAST.equals(fixAST)) {
 				continue;
 			}
 
 			//Heuristic: Do not insert a return statement on a func whose return type is void
 			//Heuristic: Do not insert a return statement in a constructor
-			if(fixAST instanceof ReturnStatement){
-				if(potentiallyBuggyStmt.parentMethodReturnsVoid() ||
+			if (fixAST instanceof ReturnStatement){
+				if (potentiallyBuggyStmt.parentMethodReturnsVoid() ||
 						potentiallyBuggyStmt.isLikelyAConstructor())
 					continue;
-
-				//Heuristic: Swapping, Appending or Replacing a return stmt to the middle of a block will make the code after it unreachable
-				ASTNode parentBlock = potentiallyBuggyStmt.blockThatContainsThisStatement();
-				if(parentBlock != null && parentBlock instanceof Block) {
-					List<ASTNode> statementsInBlock = ((Block)parentBlock).statements();
-					ASTNode lastStmtInTheBlock = statementsInBlock.get(statementsInBlock.size()-1);
-					if(!lastStmtInTheBlock.equals(faultAST)){
-						continue;
-					}
-				} else {
-					continue;
-				}
-
```

Comment:
I just noticed that this was here before and was my initial idea; I guess the assumption of no floating blocks was not a good one